python/tskit/vcf.py

```diff
@@ -0,0 +1,87 @@
+#
+# MIT License
+#
+# Copyright (c) 2019 Tskit Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Convert tree sequences to VCF.
+"""
+import math
+import time
+
+import numpy as np
+
+import _tskit
+
+
+def write_vcf(ts, output, ploidy=1, contig_id="1"):
+
+    n = ts.get_sample_size() // ploidy
+    sample_names = ["msp_{}".format(j) for j in range(n)]
+    last_pos = 0
+    positions = []
+    for variant in ts.variants():
+        pos = int(round(variant.position))
+        if pos <= last_pos:
+            pos = last_pos + 1
+        positions.append(pos)
+        last_pos = pos
```

Comment:
Here's a much faster way to do this bit:
```
positions = np.round(ts.tables.sites.position)
positions[1:] += np.cumsum(np.diff(positions) == 0)
```

Reply:
Nice, thanks @petrelharp.

Reply:
Whoops; I see that my proposal didn't actually do what I wanted. I see you aren't using it anyhow... but here's code that actually does the same thing as the `legacy_position_transform`:
```
def position_transform(positions, max_iter = 100):
    positions = np.round(positions).astype("int")
    for _ in range(max_iter):
        a = np.where(np.diff(positions) == 0)[0] + 1
        if len(a) == 0:
            break
        positions[a] += 1

# test it
def gen_positions(n, m):
    out = m * np.random.random(n)
    out.sort()
    return out

for _ in range(100):
    p1 = gen_positions(10, 20)
    p2 = p1.copy()
    transform(p1)
    legacy_position_transform(p2)
    assert(np.all(p1 == p2))
```

Reply:
Hmm, that still doesn't do exactly the same thing. It doesn't matter anyway --- this is legacy behaviour, and for large n the time spent doing this is trivial.